### ISSUE TITLE
Type of `HPDCACHE_VICTIM_SEL`

### ISSUE THREAD

**Issue by dgptha** [openhwgroup/cv-hpdcache#15]
https://github.com/openhwgroup/cv-hpdcache/blob/645e4222c3d23fbedb5b0fec1922f72fd692a40a/rtl/src/hpdcache_pkg.sv#L76C1-L82C89

The type of the `HPDCACHE_VICTIM_SEL` parameter defined in line 82 of rtl/src/hpdcache_pkg.sv should be `hpdcache_victim_sel_policy_t` and not `int unsigned`.
Having it as `int unsigned` fails on Questa (version 2019.4) when used in rtl/src/hpdcache_victim_sel.sv line 49 as the types of `HPDCACHE_VICTIM_SEL` and `HPDCACHE_VICTIM_RANDOM` do not match (`int unsigned` vs `hpdcache_victim_sel_policy_t`).

PS: problem found when using CVA6 v5.0.1.

**Comment by cfuguet** [comment]
Thank you @dgptha for reporting this issue.

I knew about the inconsistency but I did not knew that this would generate an issue in some tools (or specific versions). I actually use QuestaSim to validate the HPDcache but more recent versions.

Currently the issue is that the type hpdcache_victim_sel_policy_t is defined in the hpdcache_pkg but the actual parameter value comes from the system specific hpdcache_params_pkg. If I define the PARAM_VICTIM_SEL parameter as hpdcache_victim_sel_policy_t in the hpdcache_params_pkg, there is a circular dependency between packages. This is not valid. That is why, I defined it as a integer into the hpdcache_params_pkg.

I will try a different approach to fix the issue.
